boards/native/native_posix/Kconfig

```
@@ -25,6 +25,11 @@ config BOARD_NATIVE_POSIX_64
 	  An interrupt controller, timer (system tick), and redirects kernel prints to
 	  stdout.
 
+config BOARD_NATIVE_POSIX_NATIVE_64
+	bool
+	select BOARD_NATIVE_POSIX
+	select 64BIT
```

If using 64bit as a variant, then this should be done like this:
```suggestion
config BOARD_NATIVE_POSIX
	select 64BIT if BOARD_NATIVE_POSIX_NATIVE_64
```
at this location:
https://github.com/zephyrproject-rtos/zephyr/blob/353f45b0ec205c28c37444182b40e479f521f24e/boards/native/native_posix/Kconfig#L3

that will give us the built-in Kconfig benefits in case board / soc / variant is ever renamed and ensures names are in sync.

As well as fewer lines 😉